TrkPatRec/src/KalFinalFit_module.cc

```diff
@@ -320,10 +320,15 @@ namespace mu2e
 
 	  if(_result.missingHits.size() > 0){
 	    _kfit.addHits(srep,detmodel,_result,_maxaddchi);
-	  }else if (_cprmode){
-	    int last_iteration  = -1;
-	    _kfit.fitIteration(detmodel,_result,last_iteration);
 	  }
+//-----------------------------------------------------------------------------
+// rarely, in close-to-the-edge cases, the refit may fail and the track fit would
+// end up in undefined state. Avoid that. The commented out part didn't make sense anyway
+//-----------------------------------------------------------------------------
+	  // else if (_cprmode){
+	  //   int last_iteration  = -1;
+	  //   _kfit.fitIteration(detmodel,_result,last_iteration);
+	  // }
```

Comment:
Can we simply remove _cprmode from KalFinalFit?  This is the 2nd time it has created inconsistencies, and I don't see the benefit.

Reply:
this is not the second time, but just a proper fix to the [first] problem which you're referring to, and protection against which you had recently added. With this fix, the protection becomes unnecessary. This change is a step in the direction of taking out unnecessary differences between the fit configuration, and I propose we just do it step by step.

Reply:
Is the commented out code a strong candidate for being restored soon?  If not, just remove it.